pkgs/dart_mcp/lib/src/api/completions.dart

```diff
@@ -27,11 +27,22 @@ extension type CompleteRequest.fromMap(Map<String, Object?> _value)
   ///
   /// In the case of a [ResourceReference], it must refer to a
   /// [ResourceTemplate].
-  Reference get ref => _value['ref'] as Reference;
+  Reference get ref {
+    final ref = _value['ref'];
+    if (ref == null) {
+      throw ArgumentError('Missing ref field in $CompleteRequest.');
+    }
+    return ref as Reference;
+  }
 
   /// The argument's information.
-  CompletionArgument get argument =>
-      (_value['argument'] as Map).cast<String, Object?>() as CompletionArgument;
+  CompletionArgument get argument {
+    final argument = _value['argument'] as Map?;
+    if (argument == null) {
+      throw ArgumentError('Missing argument field in $CompleteRequest.');
+    }
+    return argument.cast<String, Object?>() as CompletionArgument;
```

Comment:
I actually think we don't need the casting here, you can just do `as CompletionArgument?` above.

Reply:
Okay, removed.  I left the other deep casting on the lists (like the roots) because that is necessary.